### PR TITLE
Add Support For More TF Types

### PR DESCRIPTION
**Pull request by NihalHarish**
### Description of changes:
- Add support for any TF Type that has a TF to Numpy mapping
- Add a test to check if the mapping dict becomes stale
- A bug that was recently reported was because the assumption that tf tensors are mapped to a single numpy type is wrong.
For example, tf string can have `npdtype.kind == "U" or npdtype.kind == "O" or npdtype.kind == "S"` ( unicode, object, str).

- The user does not control this low level detail and is determined by factors like training mode, dataset et cetera.
- This PR introduces a test that is run with the `_NP_TO_TF` datastructure from `tensorflow.python.framework.dtypes` 
- It will break if a new numpy datatype support is added, prompting the smdebug project to also sync
#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smdebug/core/tfevent/util.py [hunk 1]
```diff
@@ -10,6 +11,9 @@
 
 logger = get_logger()
 
+# TF Implements a Custom Numpy Datatype for Brain Floating Type
+_np_bfloat16 = _pywrap_bfloat16.TF_bfloat16_type()
```
Comment (vandanavk) [inline smdebug/core/tfevent/util.py:15]: is there a specific requirement for this?

File: tests/tensorflow2/test_tensorflow2_datatypes.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Third Party
+from tensorflow.python.framework.dtypes import _NP_TO_TF
+
+# First Party
+from smdebug.core.tfevent.util import _get_proto_dtype
+
+
+def test_tensorflow2_datatypes():
```
Comment (vandanavk) [inline tests/tensorflow2/test_tensorflow2_datatypes.py:8]: apart from this unit test, is there a training script that makes use of these different datatypes, that we can use for testing?

Comment (NihalHarish) [reply tests/tensorflow2/test_tensorflow2_datatypes.py:8]: The this test tracks all possible TF type to numpy types.

Any update to the mapping will make this test break, effectively helping us track possible tf to numpy type conversation.

File: smdebug/core/tfevent/util.py [hunk 3]
```diff
@@ -27,12 +27,25 @@
...
 
 def _get_proto_dtype(npdtype):
-    if npdtype.kind == "U":
+    if npdtype.kind == "U" or npdtype.kind == "O" or npdtype.kind == "S":
         return (False, "DT_STRING")
+    try:
+        from tensorflow.python import _pywrap_bfloat16
+
+        # TF Implements a Custom Numpy Datatype for Brain Floating Type
+        _np_bfloat16 = _pywrap_bfloat16.TF_bfloat16_type()
```
Comment (NihalHarish) [inline smdebug/core/tfevent/util.py:45]: There is no special ask for this, but the goal is to support every possible TF type.

File: smdebug/core/tfevent/util.py (hunk 1 above)
Comment (NihalHarish) [reply smdebug/core/tfevent/util.py]: There is no special ask for this, but the goal is to support every possible TF type.

File: tests/tensorflow2/test_tensorflow2_datatypes.py (hunk 2 above)
Comment (Vikas-kum) [reply tests/tensorflow2/test_tensorflow2_datatypes.py:9]: I guess we should also test, if these new types tensors are saved they are saved without errors. 
You should create tensors and may be try using save_tensor API

File: tests/tensorflow2/test_tensorflow2_datatypes.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Third Party
+import numpy as np
+from tensorflow.python.framework.dtypes import _NP_TO_TF
+
+# First Party
+from smdebug.core.tfevent.util import _get_proto_dtype
+
+
+def test_tensorflow2_datatypes():
```
Comment (vandanavk) [inline tests/tensorflow2/test_tensorflow2_datatypes.py:9]: add this test to TF1 as well. since common code is being changed
